packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1173,46 +996,59 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
-        if (currentRoundId <= priceRequest.lastVotingRound) return false;
-
-        // If the request has been previously resolved, return true.
-        if (priceRequest.pendingRequestIndex == UINT64_MAX) return true;
-
-        // Else, check if the price can be resolved.
-        (bool isResolvable, int256 resolvedPrice) =
-            voteInstance.resultComputation.getResolvedPrice(_computeGat(priceRequest.lastVotingRound));
-
-        // If it's not resolvable return false.
-        if (!isResolvable) return false;
-
-        // Else, the request is resolvable. Remove the element from the pending requests and update pendingRequestIndex
-        // within the price request struct to make the next entry into this method a no-op for this request.
-        _removeRequestFromPendingPriceRequests(priceRequest.pendingRequestIndex);
-
-        priceRequest.pendingRequestIndex = UINT64_MAX;
-        emit PriceResolved(
-            priceRequest.lastVotingRound,
-            priceRequest.priceRequestIndex,
-            priceRequest.identifier,
-            priceRequest.time,
-            priceRequest.ancillaryData,
-            resolvedPrice
-        );
-        return true;
+    function _resolveResolvablePriceRequests(uint256 maxTraversals) private {
+        uint32 currentRoundId = uint32(getCurrentRoundId());
+
+        uint256 index = rounds[currentRoundId].resolvedIndex;
+        uint256 requestsTraversed = 0;
+        while (index < pendingPriceRequestsIds.length && requestsTraversed < maxTraversals) {
+            ++requestsTraversed;
+
+            PriceRequest storage request = priceRequests[pendingPriceRequestsIds[index]];
+            if (request.lastVotingRound >= currentRoundId) {
+                ++index;
+                continue;
+            }
+            VoteInstance storage voteInstance = request.voteInstances[request.lastVotingRound];
+            (bool isResolvable, int256 resolvedPrice) =
+                voteInstance.resultComputation.getResolvedPrice(_computeGat(request.lastVotingRound));
+            if (!isResolvable) {
+                request.rollCount += currentRoundId - request.lastVotingRound;
+                request.lastVotingRound = currentRoundId;
+                emit PriceRequestRolled(request.identifier, request.time, request.ancillaryData, request.rollCount);
+                if (request.rollCount >= deleteAfterRollCount && !request.isGovernance) {
+                    emit PriceRequestDeleted(request.identifier, request.time, request.ancillaryData);
+
+                    delete priceRequests[pendingPriceRequestsIds[index]];
+                    _removeRequestFromPendingPriceRequestsIds(request.pendingRequestIndex);
+                } else ++index;
+
+                continue;
+            }
+
+            resolvedPriceRequestIds.push(pendingPriceRequestsIds[index]);
+
+            _removeRequestFromPendingPriceRequestsIds(request.pendingRequestIndex);
+            request.pendingRequestIndex = UINT64_MAX;
```

Comment:
Two questions:
1. Is this ever read once it's in the resolved requests array?
2. Is this variable ever used, at all? Do we need pendingRequestIndex?

Reply:
great call out! I was able to totally remove this from the data structure with 0 implications really. simplified things more and decreased storage cost.